pandas/io/common.py

```diff
@@ -805,6 +805,19 @@ def __init__(
         # _PathLike[str]], IO[bytes]]"
         super().__init__(file, mode, **kwargs_zip)  # type: ignore[arg-type]
 
+    # GH39465
+    # If an explicit archive_name is not given, we still want the file _inside_ the zip
+    # file _not_ to be named something.zip, because that causes massive confusion.
+    def infer_filename(self):
+        if isinstance(self.filename, (os.PathLike, str)):
```

Comment:
where is `.filename` actually defined?

Reply:
good question.
In `zipfile.ZipFile.__init__()`:
https://github.com/python/cpython/blob/3.10/Lib/zipfile.py#L1234

And, looking there, it looks like we need to also care of the case where a `PathLike` is passed, in which case only `self.file` and not `self.filename` are set. Will adapt to that!

Reply:
ok in the super class